typescript/infra/scripts/check/check-validator-version.ts

```diff
@@ -0,0 +1,206 @@
+  '0x0d4c1394a255568ec0ecd11795b28d1bda183ca4': 'tessellated',
+  '0x3da4ee2801ec6cc5fad73dbb94b10a203adb3d9e': 'enigma',
+  '0x4df6e8878992c300e7bfe98cac6bf7d3408b9cbf': 'imperator',
+  '0x14adb9e3598c395fe3290f3ba706c3816aa78f59': 'flow foundation',
+};
+
+type ValidatorInfo = {
+  chain: ChainName;
+  validator: string;
+  alias: string;
+  version: string;
+};
+
+function sortValidatorInfo(a: ValidatorInfo, b: ValidatorInfo) {
+  // First sort by alias
+  if (a.alias && !b.alias) return -1;
+  if (!a.alias && b.alias) return 1;
+  if (a.alias && b.alias) {
+    const aliasCompare = a.alias.localeCompare(b.alias);
+    if (aliasCompare !== 0) return aliasCompare;
+  }
+  // Then sort by validator address
+  return a.chain.localeCompare(b.chain);
+}
+
+async function main() {
+  const { environment, chains, showUpdated } = await withChains(getArgs())
+    .describe(
+      'show-updated',
+      'If enabled, prints a table with all updated validators',
+    )
+    .boolean('show-updated')
+    .default('show-updated', false).argv;
+
+  const config = getEnvironmentConfig(environment);
+  const { core } = await getHyperlaneCore(environment);
+
+  const targetNetworks = (
+    chains && chains.length > 0 ? chains : config.supportedChainNames
+  ).filter(isEthereumProtocolChain);
+
+  const mismatchedValidators: ValidatorInfo[] = [];
+  const upgradedValidators: ValidatorInfo[] = [];
+
+  // Manually add validator announce for OG Lumia chain deployment
+  const lumiaValidatorAnnounce = ValidatorAnnounce__factory.connect(
+    '0x989B7307d266151BE763935C856493D968b2affF',
+    core.multiProvider.getProvider('lumia'),
+  );
+
+  await Promise.all(
+    targetNetworks.map(async (chain) => {
+      const validatorAnnounce =
+        chain === 'lumia'
+          ? lumiaValidatorAnnounce
+          : core.getContracts(chain).validatorAnnounce;
+      const expectedValidators = defaultMultisigConfigs[chain].validators || [];
+      const storageLocations = await validatorAnnounce[
+        'getAnnouncedStorageLocations(address[])'
+      ](expectedValidators);
+
+      // For each validator on this chain
+      for (let i = 0; i < expectedValidators.length; i++) {
+        const validator = expectedValidators[i];
+        const location = storageLocations[i][0];
+        const alias =
+          KNOWN_VALIDATOR_ADDRESSES[validator.toLowerCase()] ?? 'UNKNOWN';
+
+        // Get metadata from each storage location
+        try {
+          const s3Validator = await S3Validator.fromStorageLocation(location);
+          const metadata = await s3Validator.getMetadata();
+          const gitSha = metadata?.git_sha;
+
+          if (Object.keys(acceptableValidatorVersions).includes(gitSha)) {
+            upgradedValidators.push({
+              chain,
+              validator,
+              alias,
+              version: acceptableValidatorVersions[gitSha],
+            });
+          } else {
+            mismatchedValidators.push({
```

Comment:
doesn't have to be now, we can add this later, but it would be nice to know for an unknown version how recent it is

e.g. if there's a way to know that the 1.0.0 commit is some ancestor commit of the current one, or the approx date of the commit, etc
I understand we may not have this data always, especially in cases like tesselated